src/tools/r2rdump/UnwindInfo.cs

```diff
@@ -108,12 +132,112 @@ public override string ToString()
             sb.AppendLine($"{tab}{tab}------------------");
             for (int i = 0; i < CountOfUnwindCodes; i++)
             {
-                sb.Append(UnwindCode[i].ToString());
+                sb.Append(GetUnwindCode(ref i));
+                sb.AppendLine($"{tab}{tab}------------------");
             }
             sb.AppendLine($"{tab}PersonalityRoutineRVA: 0x{PersonalityRoutineRVA:X8}");
             sb.AppendLine($"{tab}Size: {Size} bytes");
 
             return sb.ToString();
         }
+
+        private string GetUnwindCode(ref int i)
+        {
+            StringBuilder sb = new StringBuilder();
+            string tab2 = new string(' ', 8);
+
+            sb.AppendLine($"{tab2}CodeOffset: 0x{UnwindCode[i].CodeOffset:X2}");
+            sb.AppendLine($"{tab2}UnwindOp: {UnwindCode[i].UnwindOp}({(byte)UnwindCode[i].UnwindOp})");
+
+            switch (UnwindCode[i].UnwindOp)
+            {
+                case UnwindOpCodes.UWOP_PUSH_NONVOL:
+                    sb.AppendLine($"{tab2}OpInfo: {(Registers)UnwindCode[i].OpInfo}({UnwindCode[i].OpInfo})");
+                    break;
+                case UnwindOpCodes.UWOP_ALLOC_LARGE:
+                    sb.Append($"{tab2}OpInfo: {UnwindCode[i].OpInfo} - ");
+                    if (UnwindCode[i].OpInfo == 0)
+                    {
+                        i++;
+                        sb.AppendLine("Scaled small");
+                        uint frameOffset = UnwindCode[i].FrameOffset * 8;
+                        sb.AppendLine($"{tab2}FrameOffset: {UnwindCode[i].FrameOffset} * 8 = {frameOffset} = 0x{frameOffset:X5})");
+                    }
+                    else if (UnwindCode[i].OpInfo == 1)
+                    {
+                        i++;
+                        sb.AppendLine("Unscaled large");
+                        uint frameOffset = UnwindCode[i].FrameOffset;
+                        sb.AppendLine($"{tab2}FrameOffset: {UnwindCode[i].FrameOffset} * 8 = {frameOffset} = 0x{frameOffset:X8})");
+                        i++;
```

Comment:
I cannot convince myself that this `i++` is correct here when you read and assign both 16 bit components of the FrameOffset into single UnwindCode at line 72. The dumping code here seems to assume that the FrameOffset is still split between two UnwindCode entries. So it seems that when you create the Unwind code, you probably want to always extract just 16 bits of FrameOffset.

Reply:
When it reads both components of FrameOffset, the second component still gets saved as the next UnwindCode, so the i++ here skips over the second UnwindCode. 

Should it only only 16 bits and form the 32bit FrameOffset with bitshift during printing?

Reply:
> When it reads both components of FrameOffset, the second component still gets saved as the next UnwindCode

Doesn't the read at line 72 move the offset over the high component of the FrameOffset, which would mean that the next UnwindCode would not contain it?

Reply:
That's true. Should I make it so that it only increases offset by 16bits, or remove the i++?

Reply:
ok thanks!